src/main/java/hudson/remoting/Engine.java

```diff
@@ -403,6 +463,101 @@ public static Engine current() {
+        Map<String, String> properties = AccessController.doPrivileged(
+                new PrivilegedExceptionAction<Map<String, String>>() {
+                    public Map<String, String> run() throws Exception {
+                        Map<String, String> result = new HashMap<String, String>();
+                        result.put("trustStore", System.getProperty("javax.net.ssl.trustStore"));
+                        result.put("javaHome", System.getProperty("java.home"));
+                        result.put("trustStoreType",
+                                System.getProperty("javax.net.ssl.trustStoreType", KeyStore.getDefaultType()));
+                        result.put("trustStoreProvider", System.getProperty("javax.net.ssl.trustStoreProvider", ""));
+                        result.put("trustStorePasswd", System.getProperty("javax.net.ssl.trustStorePassword", ""));
+                        return result;
+                    }
+                });
+        KeyStore keystore = null;
+
+        FileInputStream trustStoreStream = null;
+        try {
+            String trustStore = properties.get("trustStore");
+            if (!"NONE".equals(trustStore)) {
+                File trustStoreFile;
+                if (trustStore != null) {
+                    trustStoreFile = new File(trustStore);
+                    trustStoreStream = getFileInputStream(trustStoreFile);
+                } else {
+                    String javaHome = properties.get("javaHome");
+                    trustStoreFile = new File(
+                            javaHome + File.separator + "lib" + File.separator + "security" + File.separator
+                                    + "jssecacerts");
+                    if ((trustStoreStream = getFileInputStream(trustStoreFile)) == null) {
+                        trustStoreFile = new File(
+                                javaHome + File.separator + "lib" + File.separator + "security" + File.separator
+                                        + "cacerts");
+                        trustStoreStream = getFileInputStream(trustStoreFile);
+                    }
+                }
+
+                if (trustStoreStream != null) {
+                    trustStore = trustStoreFile.getPath();
+                } else {
+                    trustStore = "No File Available, using empty keystore.";
+                }
+            }
+
+            String trustStoreType = properties.get("trustStoreType");
+            String trustStoreProvider = properties.get("trustStoreProvider");
+            LOGGER.log(Level.FINE, "trustStore is: {0}", trustStore);
+            LOGGER.log(Level.FINE, "trustStore type is: {0}", trustStoreType);
+            LOGGER.log(Level.FINE, "trustStore provider is: {0}", trustStoreProvider);
+
+            if (trustStoreType.length() != 0) {
+                LOGGER.log(Level.FINE, "init truststore");
+
+                if (trustStoreProvider.length() == 0) {
```

Comment:
better to use StringUtils.isEmpty(). Just in case somebody changes the implementation above